decode.go

```diff
@@ -507,6 +507,15 @@ func (d *decoder) sequence(n *node, out reflect.Value) (good bool) {
 	return true
 }
 
+func (d *decoder) setMapIndex(n *node, out, k, v reflect.Value) {
+	if d.strict {
+		if out.MapIndex(k) != zeroValue {
+			d.terrors = append(d.terrors, fmt.Sprintf("line %d: field %s already set in map", n.line+1, k.String()))
```

Comment:
Shouldn't this return without setting the value? (not that it matters that much I guess)

Reply:
While it probably doesn't matter much, I agree it shouldn't set a value after it's decided that's an error. Fixing as suggested.